client/src/main/java/io/pravega/client/stream/impl/ControllerResolverFactory.java

```diff
@@ -121,10 +139,10 @@ public String getServiceAuthority() {
         @Override
         @Synchronized
         public void start(Listener listener) {
-            Preconditions.checkState(started.compareAndSet(false, true));
+            Preconditions.checkState(!started.get());
```

Comment:
This method is not supposed to be called multiple times. gRPC's default implementation (DnsNameResolver) makes the same assumption (though the API does not document this) and throws exception if start is attempted multiple times. I have followed the same pattern here.